app/repos/people.js

```diff
@@ -35,24 +36,49 @@ const { personDto } = require('../dto/person')
 
 /**
  * @param owners
- * @param transaction
+ * @param [transaction]
+ * @param  [transaction2]
  * @returns {Promise<CreatedPersonDao[]>}
+ *
  */
 const createPeople = async (owners, transaction) => {
   if (!transaction) {
     return sequelize.transaction(async (t) => createPeople(owners, t))
   }
+  const unmanagedTransaction = await sequelize.transaction()
 
   const createdPeople = []
 
   try {
     for (const owner of owners) {
-      const person = await sequelize.models.person.create({
+      const createProperties = {
         first_name: owner.firstName,
         last_name: owner.lastName,
         birth_date: owner.dateOfBirth ?? owner.birthDate,
         person_reference: createRegistrationNumber()
-      }, { transaction })
+      }
+
+      let person
+
+      try {
+        person = await sequelize.models.person.create(createProperties, { transaction: unmanagedTransaction })
+        await unmanagedTransaction.commit()
+      } catch (e) {
+        if (e instanceof UniqueConstraintError) {
+          await unmanagedTransaction.rollback()
+          const duplicatePersonReference = e.fields.person_reference
```

Comment:
Is it not slightly simpler to just reference the person_reference from line 54 rather than extract from the error object?

Reply:
Haha, thought I was being really clever there, but you're right. 😆